src/emulator/auth/operations.ts

```diff
@@ -166,6 +166,31 @@ function signUp(
     updates.passwordUpdatedAt = Date.now();
     updates.validSince = toUnixTimestamp(new Date()).toString();
   }
+  if (reqBody.mfaInfo) {
+    const mfaInfo = [];
+    for (const factor of reqBody.mfaInfo) {
+      // the node SDK validation asserts the absence of a uid on create requests prior to submitting any API requests.
+      // e.g., `Error: "uid" is not supported when adding second factors via "createUser()"`
+      assert(
+        !Object.prototype.hasOwnProperty.call(factor, "mfaEnrollmentId"),
+        "UNEXPECTED_PARAMETER : mfaEnrollmentId"
+      );
+      assert(
+        factor.phoneInfo && isValidPhoneNumber(factor.phoneInfo),
+        "INVALID_MFA_PHONE_NUMBER : Invalid format."
+      );
+      const mfaEnrollmentId = randomId(28);
+      assert(
+        state.validateMfaEnrollmentIdForCreate(mfaEnrollmentId),
+        "INVALID_MFA_ID : enrollment ID was not unique."
```

Comment:
This should just throw an internal error since we got unlucky here and got a duplicate. There's nothing the API caller can do to fix this (apart from retrying). If you feel especially charitable, you may want to generate multiple times and pick the first one available (like what we do for localIds).